### PR TITLE
add 1-28 as a supported release branch

### PR DESCRIPTION
**Pull request by tatlat**
*Issue #, if available:*

*Description of changes:*
Add EKS-D 1.28 build support
Add 1-28 as a supported release branch
Add 1-28 metrics server
Add 1-28 cluster autoscaler

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: EKSD_LATEST_RELEASES [hunk 1]
```diff
@@ -29,4 +29,7 @@ releases:
 - branch: 1-27
   number: 10
   kubeVersion: v1.27.4
+- branch: 1-28
+  number: 1
+  kubeVersion: v1.28.0-rc.0
```
Comment (jaxesn) [inline EKSD_LATEST_RELEASES:34]: i think they cut a -2 release which is 1.28.0.

File: projects/kubernetes/cloud-provider-aws/1-28/GIT_TAG [hunk 2]
```diff
@@ -0,0 +1 @@
+v1.27.1
```
Comment (jaxesn) [inline projects/kubernetes/cloud-provider-aws/1-28/GIT_TAG:1]: no 128 yet?

File: projects/kubernetes/cloud-provider-vsphere/1-28/GIT_TAG [hunk 3]
```diff
@@ -0,0 +1 @@
+v1.27.0
```
Comment (jaxesn) [inline projects/kubernetes/cloud-provider-vsphere/1-28/GIT_TAG:1]: same question, im assuming they dont have a 1.28 release?

**Review by jaxesn — CHANGES_REQUESTED** [review]
This seems pretty close

File: release/staging-build.yml [hunk 4]
```diff
@@ -357,6 +376,15 @@ batch:
           PROJECT_PATH: projects/kubernetes/cloud-provider-vsphere
           CLONE_URL: https://git-codecommit.us-west-2.amazonaws.com/v1/repos/kubernetes.cloud-provider-vsphere
           RELEASE_BRANCH: 1-27
+    - identifier: kubernetes_cloud_provider_vsphere_1_28
```
Comment (jaxesn) [inline release/staging-build.yml:379]: we dont need this component in the staging build. its not your fault its here, but would you mind adding the skip_staging_buildspec or whatever makefile var, check out one of the packages makefiles for the exact var. there are two to add.

File: projects/kubernetes/cloud-provider-vsphere/1-28/GIT_TAG (hunk 3 above)
Comment (tatlat) [reply projects/kubernetes/cloud-provider-vsphere/1-28/GIT_TAG:1]: No 1.28 for vsphere but there is one for aws, which I'll add shortly.

File: projects/kubernetes/cloud-provider-aws/1-28/GIT_TAG (hunk 2 above)
Comment (tatlat) [reply projects/kubernetes/cloud-provider-aws/1-28/GIT_TAG]: updated this one

**Review by jaxesn — APPROVED** [review]

**Pull request merged**